### PR TITLE
Initialize releases when creating remoteBookController

### PR DESCRIPTION
**Pull request by barbaravaldez**
Initialize releases when creating remote book controller to prevent undefined error.

### CODE REVIEW

**Review by lucyzhang929 — APPROVED** [review]

**Review by VasuBhog — APPROVED** [review]

**Review by Charles-Gagnon — COMMENTED** [review]
How is this being undefined? I'm guessing when url is undefined? You should update the typings appropriately for the various properties/return types to be `IRelease[] | undefined` if undefined is a valid expected value.

But in general I would say it probably makes more sense to just default the model to having an empty array instead. Doing that should fix this I think from what I see in the code since the HTTP request properly defaults to an empty array and then populates it with the result. Unless you care about knowing when you've fetched releases and there weren't any vs we haven't actually fetched anything (undefined)

**Comment by barbaravaldez** [comment]
> How is this being undefined? I'm guessing when url is undefined? You should update the typings appropriately for the various properties/return types to be `IRelease[] | undefined` if undefined is a valid expected value.
> 
> But in general I would say it probably makes more sense to just default the model to having an empty array instead. Doing that should fix this I think from what I see in the code since the HTTP request properly defaults to an empty array and then populates it with the result. Unless you care about knowing when you've fetched releases and there weren't any vs we haven't actually fetched anything (undefined)

@Charles-Gagnon , you are right. I initialized the model correctly and we are not throwing an error anymore.

File: extensions/notebook/src/book/remoteBookController.ts [hunk 1]
```diff
@@ -14,6 +14,7 @@ const assetNameRE = /([a-zA-Z0-9]+)(?:-|_)([a-zA-Z0-9.]+)(?:-|_)([a-zA-Z0-9]+).(
 
 export class RemoteBookController {
 	constructor(public model: RemoteBookDialogModel, public outputChannel: vscode.OutputChannel) {
+		this.model.releases = [];
```
Comment (Charles-Gagnon) [inline extensions/notebook/src/book/remoteBookController.ts:17]: Why not just have the model initialize releases itself? Right now the typing for the releases property is incorrect if whatever is creating it doesn't initialize the array - and there's not really any reason I can see not to just define the internal variable as `private _releases: IRelease[] = [];`

Comment (barbaravaldez) [reply extensions/notebook/src/book/remoteBookController.ts]: that's a better idea. Let me do that.

**Review by Charles-Gagnon — APPROVED** [review]

**Pull request merged**